### ISSUE TITLE
False positive ISC001 in unmatched f-string

### ISSUE THREAD

**Issue by janosh** [astral-sh/ruff#6000]
Never mind that this thing raises a SyntaxError, it also triggers [ISC001](https://beta.ruff.rs/docs/rules/single-line-implicit-string-concatenation).

```py
f"{({None: ""}).get(key, key)}"
```

**Comment by charliermarsh** [comment]
I think this is okay given that it's not syntactically valid. It's getting parsed as `f"{({None: "`, followed by `"}).get(key, key)}"` which would indeed by an implicit concatenation and is a reasonable tokenization. Happy to reconsider if you have a syntactically valid example that's causing trouble in the wild.
